product_tier_validation/models/product_template.py

```diff
@@ -0,0 +1,32 @@
+# Copyright 2021 Open Source Integrators
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class ProductTemplate(models.Model):
+    _name = "product.template"
+    _inherit = ["product.template", "tier.validation"]
+    _tier_validation_manual_config = False
+
+    @api.model
+    def create(self, vals):
+        new = super().create(vals)
+        if new.need_validation and new.state != "confirmed":
+            new.active = False
```

Comment:
IMHO, this is a little too intrusive. Maybe should at least be a module option 'Set product inactive before validation'

Reply:
IMO the Product should start inactive until is it validated, to be sure it can't be used on transactions.
Still, I now made this easy to configure, and even extend, in case you want specific "product inactivation" conditions.